### PR TITLE
[RFC] New iteration on clang format - (version 2)

### PR DESCRIPTION
**Pull request by davidlange6**
I revived the clang-format PR from a year ago. Relative to #25529  I changed AlignConsecutiveDeclarations: true in order to try to address comments from @slava77 regarding the alignment of declarations being undone  [I'm not convinced that aligning always is an improvement]

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@davidlange6 how many of these RFCs are you planning?

File: FWCore/MessageService/interface/ELstatistics.h [hunk 1]
```diff
@@ -34,98 +33,89 @@
...
-protected:
-  void clearSummary();
+      std::map<ELextendedID, StatsCount> statisticsMap() const;
 
-  void zero() override;
+      // summarization( const ELstring & sumLines, const ELstring & sumLines )
+      // from base class
 
+    protected:
+      int           tableLimit;
```
Comment (Dr15Jones) [inline FWCore/MessageService/interface/ELstatistics.h]: In this version, the 'two columns' were preserved. So the change with respect to RFC 1 is as expected.

Personally, I prefer RFC 1.

**Comment by Dr15Jones** [comment]
NOTE: using this option will make it harder for us to use refactoring tools that rewrite the "abstract syntax tree" and create completely new lines of code. When that happens, the original alignment is lost. If we require preserving the alignment, we'd have to hand edit the code after applying such refactoring tools. 
If we allow clang tidy to always reformat to a consistent manner, then apply clang tidy after running a  refactoring tool is all we'd need to do.

File: FWCore/Concurrency/interface/WaitingTaskList.h [hunk 2]
```diff
@@ -164,12 +164,12 @@ namespace edm {
     WaitNode* createNode(WaitingTask* iTask);
 
     // ---------- member data --------------------------------
-    std::atomic<WaitNode*> m_head;
+    std::atomic<WaitNode*>      m_head;
     std::unique_ptr<WaitNode[]> m_nodeCache;
     CMS_THREAD_GUARD(m_waiting) std::exception_ptr m_exceptionPtr;
-    unsigned int m_nodeCacheSize;
+    unsigned int              m_nodeCacheSize;
```
Comment (makortel) [inline FWCore/Concurrency/interface/WaitingTaskList.h]: Here the alignment is inconsistent. Maybe `clang-format` gets confused by `CMS_THREAD_GUARD`?

File: FWCore/Framework/interface/EDProducer.h [hunk 3]
```diff
@@ -62,21 +62,21 @@ namespace edm {
...
     void doPreallocate(PreallocationConfiguration const&) {}
     void doBeginJob();
     void doEndJob();
     void doBeginRun(RunPrincipal const& rp, EventSetup const& c, ModuleCallingContext const* mcc);
     void doEndRun(RunPrincipal const& rp, EventSetup const& c, ModuleCallingContext const* mcc);
     void doBeginLuminosityBlock(LuminosityBlockPrincipal const& lbp,
-                                EventSetup const& c,
-                                ModuleCallingContext const* mcc);
+                                EventSetup const&               c,
+                                ModuleCallingContext const*     mcc);
```
Comment (makortel) [inline FWCore/Framework/interface/EDProducer.h]: Just curious, is there a way to get the `const`s aligned as well?

**Review by makortel — COMMENTED** [review]
I have mixed feelings on this suggestion. I agree there are places where the alignment would make the code clearer to read, but maybe indeed forcing alignment everywhere is too much (i.e. there are more cases where it makes the code more "strange").

File: FWCore/Concurrency/interface/WaitingTaskList.h (hunk 2 above)
Comment (davidlange6) [reply FWCore/Concurrency/interface/WaitingTaskList.h]: Right - it must not thing of CMS_THREAD_GUARD(blah..) as a declaration

File: FWCore/Framework/interface/EDProducer.h (hunk 3 above)
Comment (davidlange6) [reply FWCore/Framework/interface/EDProducer.h]: I don't see one

File: FWCore/Catalog/interface/SiteLocalConfig.h [hunk 4]
```diff
@@ -22,35 +22,35 @@ namespace edm {
...
-    virtual unsigned int const* sourceTTreeCacheSize() const = 0;
-    virtual unsigned int const* sourceTimeout() const = 0;
-    virtual bool enablePrefetching() const = 0;
-    virtual unsigned int debugLevel() const = 0;
+    virtual std::string const dataCatalog(void) const = 0;
+    virtual std::string const fallbackDataCatalog(void) const = 0;
+    virtual std::string const lookupCalibConnect(std::string const& input) const = 0;
+    virtual std::string const rfioType(void) const = 0;
+
+    virtual std::string const*              sourceCacheTempDir() const = 0;
```
Comment (VinInn) [inline FWCore/Catalog/interface/SiteLocalConfig.h]: pretty ugly indeed...

File: FWCore/Concurrency/interface/SerialTaskQueueChain.h [hunk 5]
```diff
@@ -68,98 +63,92 @@ namespace edm {
...
     std::exception_ptr ptr;
-    auto waitTaskPtr = waitTask.get();
-    push([waitTaskPtr, iAction,&ptr](){
-      //must wait until exception ptr would be set
-      auto dec = [](tbb::task* iTask){ iTask->decrement_ref_count();};
-      std::unique_ptr<tbb::task, decltype(dec)> sentry(waitTaskPtr,dec);
+    auto               waitTaskPtr = waitTask.get();
+    push([waitTaskPtr, iAction, &ptr]() {
+      // must wait until exception ptr would be set
+      auto                                      dec = [](tbb::task* iTask) { iTask->decrement_ref_count(); };
```
Comment (VinInn) [inline FWCore/Concurrency/interface/SerialTaskQueueChain.h]: wonderful!

**Comment by davidlange6** [comment]
looks like RFC1 was the more interesting choice.

**Pull request closed without merging**